api/src/pcapi/core/offers/api.py

```diff
@@ -450,43 +451,42 @@ def update_offer(
 
 def batch_update_offers(query: BaseQuery, update_fields: dict, send_email_notification: bool = False) -> None:
     query = query.filter(models.Offer.validation == models.OfferValidationStatus.APPROVED)
-    raw_results = query.with_entities(models.Offer.id, models.Offer.venueId).all()
-    offer_ids: typing.Sequence[int] = []
-    venue_ids: typing.Sequence[int] = []
-    if raw_results:
-        offer_ids, venue_ids = zip(*raw_results)
-    venue_ids = sorted(set(venue_ids))
-    number_of_offers_to_update = len(offer_ids)
-    logger.info(
-        "Batch update of offers",
-        extra={"updated_fields": update_fields, "nb_offers": number_of_offers_to_update, "venue_ids": venue_ids},
-    )
-    if "isActive" in update_fields.keys():
-        message = "Offers has been activated" if update_fields["isActive"] else "Offers has been deactivated"
-        technical_message_id = "offers.activated" if update_fields["isActive"] else "offers.deactivated"
-        logger.info(
-            message,
-            extra={"offer_ids": offer_ids, "venue_ids": venue_ids},
-            technical_message_id=technical_message_id,
-        )
+    query = query.with_entities(models.Offer.id, models.Offer.venueId).yield_per(2_500)
+
+    offers_count = 0
+    found_venue_ids = set()
+
+    logger.info("Batch update of offers: start", extra={"updated_fields": update_fields})
+
+    for chunk in get_chunks(query, chunk_size=2_500):
+        raw_offer_ids, raw_venue_ids = zip(*chunk)
+        offer_ids = set(raw_offer_ids)
+        venue_ids = set(raw_venue_ids)
 
-    batch_size = 1000
-    for current_start_index in range(0, number_of_offers_to_update, batch_size):
-        offer_ids_batch = offer_ids[
-            current_start_index : min(current_start_index + batch_size, number_of_offers_to_update)
-        ]
+        offers_count += len(offer_ids)
+        found_venue_ids |= set(venue_ids)
 
-        query_to_update = models.Offer.query.filter(models.Offer.id.in_(offer_ids_batch))
+        query_to_update = models.Offer.query.filter(models.Offer.id.in_(offer_ids))
         query_to_update.update(update_fields, synchronize_session=False)
         if is_managed_transaction():
             db.session.flush()
         else:
             db.session.commit()
```

Comment:
J'ai testé la fonction en staging, elle buggue dans le cas d'un commit car elle perd le curseur dans ce cas là. 

Du coup dans la boucle for, il ne faut faire que des `db.session.flush()` et déplacer ce bloc de code à la fin de la fonction. 

```suggestion
        db.session.flush()
```

Reply:
Dans ce cas, l'appel à `db.session.flush()` dans ce bloc final n'est plus nécessaire, si ?

Reply:
Ah oui tu as raison